graylog2-server/src/main/java/org/graylog2/inputs/InputServiceImpl.java

```diff
@@ -142,6 +142,9 @@ public Input create(Map<String, Object> fields) {
 
     @Override
     public Input find(String id) throws NotFoundException {
+        if (!ObjectId.isValid(id)) {
+            throw new NotFoundException("Input id <" + id + "> is invalid!");
```

Comment:
I don't think so. This should be a pretty high-level abstraction of the Input persistence. From what the consumer of it needs to know is that there is no entity with that id, not that the id is misconstructed, which does not help the consumer of this method in any way. The id which is invalid for this implementation could be perfectly valid for a different implementation anyhow.